File: server/piwars-common/src/main/java/org/ah/piwars/virtualrover/message/ServerRequestScreenshotMessage.java
package org.ah.piwars.virtualrover.message;

import org.ah.themvsus.engine.common.message.Message;
import org.ah.themvsus.engine.common.message.MessageCode;
import org.ah.themvsus.engine.common.message.MessageFactory;
import org.ah.themvsus.engine.common.transfer.Serializer;

import static org.ah.piwars.virtualrover.message.PiWarsMessageCode.ServerRequestScreenshot;

public class ServerRequestScreenshotMessage extends Message {

    private int cameraId;

    public ServerRequestScreenshotMessage(MessageFactory messageFactory) {
        super(messageFactory);
    }

    @Override
    public MessageCode getType() { return ServerRequestScreenshot; }

    @Override
    public int size() {
        return super.size() + 2;
    }

    public void setCameraId(int cameraId) {
        this.cameraId = cameraId;
    }

    public int getCameraId() {
        return cameraId;
    }

    @Override
    protected void deserializeImpl(Serializer deserializer) {
        super.deserializeImpl(deserializer);
        cameraId = deserializer.deserializeUnsignedShort();
    }

    @Override
    protected void serializeImpl(Serializer serializer) {
        serializer.serializeUnsignedShort(cameraId);
    }
}

File: fishtank/server/fishtank-python/src/fishtankclient/engine/message/ServerRequestScreenshotMessage.py
from fishtankclient.engine.message.Message import Message


class ServerRequestScreenshotMessage(Message):
    def __init__(self, factory, message_type):
        super(ServerRequestScreenshotMessage, self).__init__(factory, message_type)
        self._camera_id = 0

    def get_camera_id(self):
        return self.camera_id

    def set_camera_id(self, camera_id):
        self._camera_id = camera_id

    def size(self):
        return super(ServerRequestScreenshotMessage, self).size() + 2

    def deserialize_impl(self, deserializer):
        # super(FrameMessage, self).deserialize_impl(deserializer)
        self._camera_id = deserializer.deserialize_short()

    def serialize_impl(self, serializer):
        # super(FrameMessage, self).serialize_impl(serializer)
        serializer.serialize_unsigned_short(self._camera_id)
